lib/mongodb.js

```diff
@@ -67,7 +67,11 @@ exports.initialize = function initializeDataSource(dataSource, callback) {
   dataSource.ObjectID = mongodb.ObjectID;
 
   if (callback) {
-    dataSource.connector.connect(callback);
+    if (s.lazyConnect) {
+      callback(null);
```

Comment:
We should wrap it inside `process.nextTick()` or `setImmediate()` so that the callback happens after the function returns.